### PR TITLE
DASH-239 cursor implementation

### PR DESCRIPTION
**Pull request by ivanovSPvirtru**
Work in progress

### CODE REVIEW

File: auditexport/cli.py [hunk 1]
```diff
@@ -46,12 +46,18 @@ def main():
...
+                        dest='useCursor',
                         default=False,
                         required=False,
                         action='store_true')
+    parser.add_argument('-l', '--limit',
+                        help='Number of records we pull for each chunk when we use --cursor. Default is 100',
+                        dest='limit',
+                        default=100,
+                        required=False,
+                        action='store_true')
```
Comment (ivanovSPvirtru) [inline auditexport/cli.py]: Screen Shot 2019-03-12 at 19 45 21

@Nnamdi101 exception looks strange)) Have any ideas?

Comment (Nnamdi101) [reply auditexport/cli.py]: remove `action='store_true` from limit.  That is used only when you want to specify the param as a true/false param. For example adding `-v` option automatically sets the verbose param to true. No need to do `-v=true`.

File: auditexport/cli.py [hunk 2]
```diff
@@ -117,40 +124,46 @@ def process(args, auditclient, utils):
...
+            'sort': 'timestamp:asc',
         }
     }
 
-    if(nextPageStartKey and useBookMark):
-        req['query']['nextPageStartKey'] = nextPageStartKey
+    if(nextPageCursor and useCursor):
+        req['query']['nextPageCursor'] = nextPageCursor
+
+    if(useCursor):
```
Comment (Nnamdi101) [inline auditexport/cli.py]: should this be `if(limit)`?

Comment (ivanovSPvirtru) [reply auditexport/cli.py]: Yeah. Probably it should be `if(limit or cursor)` cause if we use cursor we should set limit (or we won't split request on chunks)

**Review by Nnamdi101 — APPROVED** [review]

**Comment by svirtryu** [comment]
@ivanovSPvirtru I have updated this to use a 2.x-RC branch to prepare for release when it's ready. can you please update changelog with a major version change and details of the changes? When that's done we should be able to merge and close out the ticket.

File: README.md [hunk 3]
```diff
@@ -48,15 +48,16 @@ to export audit records to csv, provide a path to a folder
 to export audit records to syslog, provide a host and a port
 ### `--sysloghost=0.0.0.0 --syslogport=514`
 
-to pull records since the last time the script was run, set the bookmark option
-### `--bookmark` or `-b`
-you can also provide a `bookmark.ini` file, in  the `.auditexport` directory, which will tell the script where to start pulling records
+to pull records since the last time the script was run, set the cursor option
+### `--cursor` or `-c`
```
Comment (svirtryu) [inline README.md:52]: Note in changelog this breaking change.

Comment (ivanovSPvirtru) [reply README.md:52]: done

**Pull request merged**